### PR TITLE
SA 2D axisymmetric source terms

### PR DESCRIPTION
**Pull request by Corentinthomee**
## Proposed Changes
*Give a brief overview of your contribution here in a few sentences.*
Additional source terms in the SA turbulence model have been added for the axisymmetric formulation.

In the baseline model, the only term that has an impact is:
image
Rewriting it in cylindrical coordinates, an additional source term is obtained:
image

In the Negative variant of the model, this source varies slightly due to the presence of $f_n$:
image

The compressibility correction also requires a new source. Rewriting
image
in cylindrical coordinates yields the new source
image

For the other model variants, it looks like the modification of strain magnitude (done for SST) takes care of the differences, but I might be mistaken. 

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*
#1565 

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [X] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [X] I used the pre-commit hook to prevent dirty commits and used `pre-commit run --all` to format old commits.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

**Comment by Corentinthomee** [comment]
I'm thinking about reusing the SST test case ([axisymmetric_rans/air_nozzle](https://github.com/su2code/SU2/tree/master/TestCases/axisymmetric_rans/air_nozzle)) so that the same mesh can be used. Would that be fine ?

**Comment by bigfooted** [comment]
> I'm thinking about reusing the SST test case ([axisymmetric_rans/air_nozzle](https://github.com/su2code/SU2/tree/master/TestCases/axisymmetric_rans/air_nozzle)) so that the same mesh can be used. Would that be fine ?

That's a quite specific testcase. How did you test the implementation? Do you have some simple results for the turbulent axisymmetric jet? We know that the results for round jets are not super good for standard SA, but at least we know what the result should look like, here on p.32 - 35 they test SA for the round jet and compare to the Wygnanski & Fiedler measurements:
https://www.researchgate.net/publication/24296213_Turbulence_Modeling_Validation_Testing_and_Development

The domain is just a rectangle so pretty simple to set up.

You could also use the V&V test that we have, but it is a variable density jet:
https://su2code.github.io/vandv/SANDIA_jet/

File: SU2_CFD/include/numerics/turbulent/turb_sources.hpp [hunk 1]
```diff
@@ -77,9 +77,41 @@ class CSourceBase_TurbSA : public CNumerics {
...
+      const su2double fn = (cn1 + Xi*Xi*Xi) / (cn1 - Xi*Xi*Xi);
+      nu_e = nu + fn * nue;
+    } else {
+      nu_e = nu + nue;
+    }
+
+    /* Diffusion source term */
+    const su2double dv_axi = (1.0/sigma)*nu_e*ScalarVar_Grad_i[0][1];
+
+    Residual += yinv * dv_axi * Volume;
```
Comment (pcarruscag) [inline SU2_CFD/include/numerics/turbulent/turb_sources.hpp:112]: Is it worth including the contribution to the Jacobian?

**Review by pcarruscag — COMMENTED** [review]
Nice work :+1: clean code

**Comment by Corentinthomee** [comment]
> > I'm thinking about reusing the SST test case ([axisymmetric_rans/air_nozzle](https://github.com/su2code/SU2/tree/master/TestCases/axisymmetric_rans/air_nozzle)) so that the same mesh can be used. Would that be fine ?
> 
> That's a quite specific testcase. How did you test the implementation? Do you have some simple results for the turbulent axisymmetric jet? We know that the results for round jets are not super good for standard SA, but at least we know what the result should look like, here on p.32 - 35 they test SA for the round jet and compare to the Wygnanski & Fiedler measurements: https://www.researchgate.net/publication/24296213_Turbulence_Modeling_Validation_Testing_and_Development
> 
> The domain is just a rectangle so pretty simple to set up.
> 
> You could also use the V&V test that we have, but it is a variable density jet: https://su2code.github.io/vandv/SANDIA_jet/

I did some tests on a supersonic jet, since I already had the files, but it's a quite complex case so it's not that good to verify the implementation. I will look into the simpler cases you mentioned !

**Comment by bigfooted** [comment]
Any update on a testcase to add to the PR?

**Comment by Corentinthomee** [comment]
Sorry, I have been quite busy the past few weeks.
I will try to get it done in the next few days.

**Comment by pcarruscag** [comment]
@Corentinthomee can you add a testcase? otherwise please update AUTHORS.md so we can merge

**Comment by bigfooted** [comment]
I will add a testcase in a separate PR.

**Pull request merged**